mycity/mycity/intents/trash_intent.py

```diff
@@ -174,8 +174,8 @@ def validate_found_address(found_address, user_provided_address):
         return False
 
     # Re-collect replaces South with S and North with N
-    found_address["street_name"] = found_address["street_name"].replace('S', "South")
-    found_address["street_name"] = found_address["street_name"].replace('N', "North")
+    found_address["street_name"] = re.sub('^S ', "South", found_address["street_name"])
+    found_address["street_name"] = re.sub('^N ', "North", found_address["street_name"])

```

Comment:
1) there's the `r''` literal for regular expression help, mostly changes how `\` escaping behavior works
2) `r'^S\.? '` for the optional abbreviation period
```suggestion
    found_address["street_name"] = re.sub(r'^N\.? ', "North", found_address["street_name"])
```

Reply:
Good suggestion, updated!